script/hardhat/fork/vaultUpgradesVip.ts

```diff
@@ -0,0 +1,339 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+
+import { IAccessControlManager, VAIVault, VRTVault, VRTVaultProxy, XVSVault, XVSVaultProxy } from "../../../typechain";
+import { forking, testVip } from "./vip-framework";
+import { ProposalType } from "./vip-framework/types";
+import { makeProposal } from "./vip-framework/utils";
+import { initMainnetUser } from "./vip-framework/utils";
+
+const ACM = "0x4788629ABc6cFCA10F9f969efdEAa1cF70c23555";
+const XVS_VAULT_PROXY = "0x051100480289e704d20e9DB4804837068f3f9204";
+const VAI_VAULT_PROXY = "0x0667Eed0a0aAb930af74a3dfeDD263A73994f216";
+const VRT_VAULT_PROXY = "0x98bF4786D72AAEF6c714425126Dd92f149e3F334";
+const XVS_OLD = "0xA0c958ca0FfA25253DE0a23f98aD3062F3987073";
+const VAI_OLD = "0x7680C89Eb3e58dEc4D38093B4803be2b7f257360";
+const VRT_OLD = "0xA3EEA5e491AD45caE30F6E0a315A275cc99EE147";
+const XVS_NEW = "0x5F0A1E8941d6d0E3d220237FA326F0bfbbCC3E9F";
+const VAI_NEW = "0x23D1F85ACDC301bA61D46Ab1c986B781a3776435";
+const VRT_NEW = "0x023EEDd63947634854654b2607088EB12B2d185f";
+const FAST_TRACK_TIMELOCK = "0x555ba73dB1b006F3f2C7dB7126d6e4343aDBce02";
+const CRITICAL_TIMELOCK = "0x213c446ec11e45b15a6E29C1C1b402B8897f606d";
+const MULTISIG = "0x1C2CAc6ec528c20800B2fe734820D87b581eAA6B";
+const NORMAL_TIMELOCK = "0x939bD8d64c0A9583A7Dcea9933f7b21697ab6396";
+
+export const vip105 = () => {
+  const meta = {
+    version: "v2",
+    title: "VIP-105 Vault Upgrades",
+    description: `
+    Accept the ownership of the XVSVault and VRTVault contracts (previously we had to execute a multisig TX to offer this change)
+      the new owner will be the Normal timelock contract (0x939bd8d64c0a9583a7dcea9933f7b21697ab6396)
+
+    Upgrade the implementations of the three vaults
+      Address of the XVSVault proxy in main net: https://bscscan.com/address/0x051100480289e704d20e9db4804837068f3f9204
+      Address of the VRTVault proxy in main net: https://bscscan.com/address/0x98bF4786D72AAEF6c714425126Dd92f149e3F334
+      Address of the VAIVault proxy in main net: https://bscscan.com/address/0x0667eed0a0aab930af74a3dfedd263a73994f216
+
+    Authorize the Fast-track and Critical timelock contracts to invoke the admin functions in the vaults, specifically for the following functions:
+      XVSVault:
+        pause
+        resume
+      VRTVault
+        pause
+        resume
+      VAIVault
+        pause
+        resume
```

Comment:
I think as a general rule we should allow normal timelock if fast-track is allowed, and allow both normal and fast-track if critical is allowed. This is to accomodate for scenarios when we want to execute multiple commands with different access level in a single VIP (e.g. upgrade VAIController and pause VAI vault).

@chechu @aditya-routv WDYT?

Reply:
I agree. But the Normal timelock is already granted by default, right? Would you add an explicit rule?

Reply:
I was wrong, the normal timelock is not granted by default. We have to be explicit with this permission. And in general I would follow the rule suggested by @kkirka

Reply:
Done